src/Transport/TestTransport.php

```diff
@@ -279,6 +282,24 @@ public function reject(Envelope $envelope): void
         $this->collectMessage(self::$rejected, $envelope);
     }
 
+    public function all(?int $limit = null): iterable
+    {
+        return \array_map(
+            fn(TestEnvelope $envelope) => $envelope->envelope,
+            \array_slice($this->queue()->all(), 0, $limit),
+        );
+    }
+
+    public function find(mixed $id): ?Envelope
+    {
+        throw new \BadMethodCallException('Not supported.');
```

Comment:
I think this is enough... any thoughts @aegypius ?

Reply:
I think it's enough too.